src/kibana/components/doc_table/components/table_row.js

```diff
@@ -187,6 +192,16 @@ define(function (require) {
         function _getValForField(row, field) {
           var val;
 
+          if (row.highlight && row.highlight[field]) {
+            // Strip out the highlight tags so we have the "original" value
+            var untagged = _.map(row.highlight[field], function (value) {
+              return value
+                .split(highlightTags.pre).join('')
+                .split(highlightTags.post).join('');
+            });
```

Comment:
This is happening because the highlight filter expect the "original" formatted value of the field, which we of course do not have, so we sort of fake it. It also allows the field formatter to apply cleanly.